### PR TITLE
[WIP] Issue #907 Pre-requisite installer for Windows

### LINKED ISSUE #907: Minishift installer for Windows
It would be great for user if we could provide `.exe` or `.msi` installer for Windows.

We can take some reference from [minikube installer for windows](https://github.com/kubernetes/minikube/issues/449).

This is not a hight priority IMO but good to have. We can take it gradually.

### PR DESCRIPTION
**Pull request by budhram**
Fixes #907  Just for review purpose.

See here https://github.com/minishift/minishift/issues/907#issuecomment-430962699

to check the pre-requisite to run `minishift setup`.

### CODE REVIEW

**Comment by gbraad** [comment]
comments at an early stage are importamt with this, as they decide a lot about the implementation. generated scripts will never pass execution policy, as they are not signed.

**Comment by budhram** [comment]
> generated scripts will never pass execution policy, as they are not signed.

Didn't know this before this comment 😄

**Comment by LalatenduMohanty** [comment]
@budhram  https://github.com/minishift/minishift/issues/907#issuecomment-432609602

File: pkg/minishift/setup/platform/platformsetup_windows.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	"path/filepath"
+	"strings"
+)
+
+const (
+	GetGroupCmd = `(New-Object System.Security.Principal.SecurityIdentifier("S-1-5-32-578")).Translate([System.Security.Principal.NTAccount]).Value`
+)
+
+// Default External Virtual Switch Name
+const ExternalVirtualSwitchName = "minishift-ext-switch"
```
Comment (LalatenduMohanty) [inline pkg/minishift/setup/platform/platformsetup_windows.go]: suggestion s/minishift-ext-switch/minishift-external/ no need to add switch to the name as it is understood.

File: pkg/minishift/setup/platform/platformsetup_windows.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		`$switchName = if ($env:HYPERV_VIRTUAL_SWITCH) {$env:HYPERV_VIRTUAL_SWITCH} else {"%s"}`,
+		`try { Get-VMSwitch -Name $switchName }`,
+		`catch { $adapterName;`,
+		`[array]$adapters = Get-NetAdapter -Physical | Where-Object { $_.status -eq "up" }`,
+		`foreach ($adapter in $adapters) {`,
+		`$adapterName = $adapter.Name`,
+		`if ($adapterName -like "ethernet*") { break } }`,
+		`New-VMSwitch -Name $switchName -NetAdapterName $adapterName`,
+		`[Environment]::SetEnvironmentVariable("HYPERV_VIRTUAL_SWITCH", $switchName, "User") }`,
+	}
```
Comment (LalatenduMohanty) [inline pkg/minishift/setup/platform/platformsetup_windows.go]: @budhram Can you please explain how you are deciding which network adapter to use for the external typer virtual switch.  The above block is not easy to understand.

File: pkg/minishift/setup/hypervisor/virtualbox_windows.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/minishift/minishift/pkg/util/filehelper"
+	"strings"
+)
+
+var (
+	vbExecutables = []string{"VirtualBox.exe", "VBoxManage.exe"}
+	vbDownloadURL = "https://download.virtualbox.org/virtualbox/5.2.20/VirtualBox-5.2.20-125813-Win.exe"
+)
+
+func detectExistingInstall() (string, error) {
```
Comment (anjannath) [inline pkg/minishift/setup/hypervisor/virtualbox_windows.go]: We already have some of this checks as part of preflight checks

Comment (budhram) [reply pkg/minishift/setup/hypervisor/virtualbox_windows.go]: @anjannath Sure. Agree, I am working on virtualbox part
and my main motto ATM is not to refactor even if need to duplicate as refactor will touch other features code base and chance of regression is high.

We can create refactor task later on.

**Comment by budhram** [comment]
Closing it

**Pull request closed without merging**